### PR TITLE
VMO-6346 Block code (name) editor behavior

### PR DESCRIPTION
**Pull request by object-required**
The `block.name` is updated when user edits either _block title_ or _block code_.

Initially, I had the idea to leave `block.name` empty until a user explicitly changes it, but it turned impractical as both `IBlock` and JSON Schema define `name` attribute mandatory.

This PR adds a new field to `block.ui_metadata.has_user_defined_name` to mark block names edited by user. The newly created `block_setName` action contains related conditions so that we don't overwrite `name` together with `label`.

See demo 🎞️ in [VMO-6346](https://viamoinc.atlassian.net/browse/VMO-6346)

### CODE REVIEW

**Review by bulatgab — APPROVED** [review]

File: src/store/flow/block.ts
Comment (safydy-r) [inline src/store/flow/block.ts:145]: @object-required , If I'm not wrong, the logic here is `if the user hasn't defined it yet, then auto update the name from the payload value`, right ?

With that, we know that the **final state** we want to know is "should we auto update the name?". What if we use that final state to rename the flag by using:
- `should_auto_update_name` instead of `has_user_defined_name`
- `block_autoUpdateName`  instead of `block_setUserDefinedName`
- `shouldAutoUpdateName` instead of `canSetName` (which is more like permission name). Or we can even avoid creating a variable for that, it's only used once.

That would be easier to read I think, your thoughts?

- [ ] rename the flag if you agree

File: src/store/flow/block.ts
Comment (safydy-r) [inline src/store/flow/block.ts:23]: Let's use the same approach to update config & vendor_metadata `by path`, eg: [block_updateConfigByPath](https://github.com/FLOIP/flow-builder/blob/master/src/store/flow/block.ts#L94).

- [ ] Create a new mutation `block_updateUiMetadataByPath` in https://github.com/FLOIP/flow-builder/blob/master/src/store/flow/block.ts, and call it here.

File: src/components/interaction-designer/block-editors/NameEditor.vue
Comment (safydy-r) [inline src/components/interaction-designer/block-editors/NameEditor.vue:81]: good catch

File: src/components/interaction-designer/block-editors/NameEditor.vue
Comment (safydy-r) [inline src/components/interaction-designer/block-editors/NameEditor.vue:94]: @object-required , can we get the type of `IBlockUIMetadata['has_user_defined_name']` if we didn't update flow-runner interface `IBlockUIMetadata` to have `has_user_defined_name` ?
Or did you add it but forgot to share the PR ?

Comment (safydy-r) [reply src/components/interaction-designer/block-editors/NameEditor.vue:94]: - [ ] update `IBlockUIMetadata` with the new flag (optional boolean)

**Review by safydy-r — CHANGES_REQUESTED** [review]
In general I'm good, left several questions/requests though.

File: src/components/interaction-designer/block-editors/NameEditor.vue
Comment (object-required) [reply src/components/interaction-designer/block-editors/NameEditor.vue]: Added to https://github.com/FLOIP/flow-runner/pull/78

File: package.json
Comment (safydy-r) [inline package.json:31]: - [ ] use the published version from npm after flow-runner approval

File: src/store/flow/block.ts
Comment (safydy-r) [inline src/store/flow/block.ts:172]: - [ ] move this  in JS Doc, above block_setName definition

File: src/store/flow/block.ts
Comment (safydy-r) [inline src/store/flow/block.ts:111]: 💯

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue
Comment (safydy-r) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:475]: - [ ] make this optional as in flow-runner recommendation https://github.com/FLOIP/flow-runner/pull/78/files#diff-8e0400228dd698ac38c3bdcccec67c81d0c45614354d8cf60f4e6af3dd863aebR52

**Review by safydy-r — APPROVED** [review]
@object-required , I'm pre-approving with one concern:
- the `should_auto_update_name` should optional I think, but let me know if you disagree

**Comment by object-required** [comment]
> @object-required , I'm pre-approving with one concern:
> 
> * the `should_auto_update_name` should optional I think, but let me know if you disagree

I see you addressed this concern in https://github.com/FLOIP/flow-runner/pull/78/commits/ba6152209f77e8da7f7b1d5fb763ad34e619bad2, thank you for taking action here 🙏

**Pull request merged**
